core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -787,6 +787,15 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
 
     val k = self._asyncContWith
 
+    // TODO: To fix?
+    //         1. `k` is `_asyncContWith` which is a `ZIO[Any, Any, Any] => Any`
+    //         2. `notBlockingOn` is a `() => FiberId`
+    //         3. `notBlockingOn` seems to never be assigned to `_asyncContWith` (needs a deeper look to be sure about this)
+    //         4. which would mean that `k ne FiberRuntime.notBlockingOn` would always be `true`
+    //         5. which would mean that this `(k ne FiberRuntime.notBlockingOn)` test is useless
+    //      Some questions to answer then:
+    //         1. Why and when was it added?
+    //         2. If it's indeed incorrect/useless, should it be removed or replaced with something else? What was the intention of the original author?
     if ((k ne null) && (k ne FiberRuntime.notBlockingOn)) {
```

Comment:
I think it's useless and can be removed. Maybe it was part of my work in (high-perfomrantly) integrating blocking into ZIO's Loom support.